### PR TITLE
docs: update readme and create documentation PE-2332

### PR DESCRIPTION
**Pull request by luqven**
TSIA

- incorporates https://github.com/imgix/web-components/pull/49

### Stacked PR Chain: PE-2332
| PR | Title | Status |  Merges Into  |
|:--:|:------|:-------|:-------------:|
|#1|👉 docs: update readme and create documentation PE-2332||-|
|#2|docs: create ix-video hosted docs PE-2332||#1|
|#3|docs: rename vuepress to docs PE-2332||#2|
|#4|docs(release): v1.0.0-rc.1 PE-2332||#3|

### CODE REVIEW

File: documentation/vue.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      source="https://assets.imgix.video/videos/girl-reading-book-in-library.mp4"
+      @error="(event) => warn('error:', event)"
+    ></ix-video>
+  </div>
+</template>
+```
+
+However, Vue might show some warnings in the console if you don't declare your custom element in it's configuration. To turn these warnings off, you can let vue that the elements that start with `ix` are custom elements.
+
+If you're using Vite, you can add the following to your `vite.config.js`:
```
Comment (sherwinski) [inline documentation/vue.md]:
```suggestion
```

File: documentation/vue.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        ...options,
+        compilerOptions: {
+          // treat any tag that starts with ix- as custom elements
+          isCustomElement: (name) => name.startsWith('ix-'),
+        }
+      }))
+  }
+```
+
+Or with Vite:
```
Comment (sherwinski) [inline documentation/vue.md]:
```suggestion
Or if you're using Vite, you can add the following to your `vite.config.js`:
```

File: documentation/vue.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    <ix-video
+      controls
+      source="https://assets.imgix.video/videos/girl-reading-book-in-library.mp4"
+      @error="(event) => warn('error:', event)"
+    ></ix-video>
+  </div>
+</template>
+```
+
+However, Vue might show some warnings in the console if you don't declare your custom element in it's configuration. To turn these warnings off, you can let vue that the elements that start with `ix` are custom elements.
```
Comment (sherwinski) [inline documentation/vue.md]:
```suggestion
You will need to signal to the Vue compiler that this component is a custom element by adding the following:
```

File: README.md [hunk 4]
```diff
@@ -1,215 +1,62 @@
...
+# Usage
 
-Linting of TypeScript files is provided by [ESLint](eslint.org) and [TypeScript ESLint](https://github.com/typescript-eslint/typescript-eslint). In addition, [lit-analyzer](https://www.npmjs.com/package/lit-analyzer) is used to type-check and lint lit-html templates with the same engine and rules as lit-plugin.
+Look at our docs to see how to us this inside your [React](/documentation/react.md), [Vue](/documentation/vue.md), or other [web-apps](/documentation/static.md).
 
-The rules are mostly the recommended rules from each project, but some have been turned off to make LitElement usage easier. The recommended rules are pretty strict, so you may want to relax them by editing `.eslintrc.json` and `tsconfig.json`.
+## Using `<ix-video>` on an HTML static page
 
-To lint the project run:
+Below is an example of how to import and use the `ix-video` custom element from a CDN on a static HTML page.
```
Comment (sherwinski) [inline README.md]:
```suggestion
Below is an example of how to import and use the `ix-video` custom element from a direct import or CDN on a static HTML page.
```

File: README.md [hunk 5]
```diff
@@ -1,215 +1,62 @@
...
-The project is setup to recommend lit-plugin to VS Code users if they don't already have it installed.
-
-## Linting
+# Usage
 
-Linting of TypeScript files is provided by [ESLint](eslint.org) and [TypeScript ESLint](https://github.com/typescript-eslint/typescript-eslint). In addition, [lit-analyzer](https://www.npmjs.com/package/lit-analyzer) is used to type-check and lint lit-html templates with the same engine and rules as lit-plugin.
+Look at our docs to see how to us this inside your [React](/documentation/react.md), [Vue](/documentation/vue.md), or other [web-apps](/documentation/static.md).
 
-The rules are mostly the recommended rules from each project, but some have been turned off to make LitElement usage easier. The recommended rules are pretty strict, so you may want to relax them by editing `.eslintrc.json` and `tsconfig.json`.
+## Using `<ix-video>` on an HTML static page
```
Comment (sherwinski) [inline README.md]: Super nit, feel free to ignore but I thought this sounded a bit clearer (and consistent with L41)
```suggestion
## Using `<ix-video>` on a static HTML page
```

File: documentation/contributing.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
+# Development
```
Comment (sherwinski) [inline documentation/contributing.md]: Personal preference but I think `CONTRIBUTING.md` should generally live in the project root

File: documentation/attributes.md [hunk 7]
```diff
@@ -0,0 +1,34 @@
+### Attributes
```
Comment (sherwinski) [inline documentation/attributes.md]: Any reason to break attributes into its own nested doc? To me it feels like something that should be on the main README for immediate reference

**Review by sherwinski — APPROVED** [review]

File: documentation/vue.md [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+    config.module
+      .rule('vue')
+      .use('vue-loader')
+      .tap(options => ({
+        ...options,
+        compilerOptions: {
+          // treat any tag that starts with ix- as custom elements
+          isCustomElement: (name) => name.startsWith('ix-'),
+        }
+      }))
```
Comment (sherwinski) [inline documentation/vue.md]: This was going to keep me up at night if I didn't say something
```suggestion
      })
    )
```

File: documentation/contributing.md (hunk 6 above)
Comment (luqven) [reply documentation/contributing.md]: 💯 that mstm

File: documentation/attributes.md (hunk 7 above)
Comment (luqven) [reply documentation/attributes.md]: No strong opinion here. Happy to move it to the main README.

**Review by sherwinski — APPROVED** [review]

File: documentation/vue.md [hunk 9]
```diff
@@ -0,0 +1,70 @@
+# Using `≤ix-video>` custom element in a Vue app
```
Comment (sherwinski) [inline documentation/vue.md]: Ah just noticed this while reviewing a different PR down the stack
```suggestion
# Using `<ix-video>` custom element in a Vue app
```

**Pull request merged**